src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -194,15 +231,15 @@ async ValueTask SynchronizeTextChangesAsync(Document oldDocument, Document newDo
             }
 
             // get text changes
-            var textChanges = newText.GetTextChanges(oldText);
-            if (textChanges.Count == 0)
+            var textChanges = newText.GetTextChanges(oldText).AsImmutable();
```

Comment:
changed the API to take an immutable array as well.

Reply:
Would something like ToImmutableArrayOrEmpty prevent recreating the array?

Reply:
That's what AsImmutable should do.  If not, we need ot unify :)

Reply:
Yes, I find it very confusing that there as so many ways to convert from IEnumerable to ImmutableArray, but I think most of them don't do the check for it already being an immutable array

Reply:
ultimately, it's the underlying array that matters.  as boxing and unboxing the immutable array doesn't actually give the same IA again.

Reply:
Oh, I didn't track it into CreateRange, I didn't think ImmutableArray had that built in. Nice!